### PR TITLE
Document changes in HNC v0.5

### PR DESCRIPTION
**Pull request by adrianludwin**
Documented the new behaviour of subnamespace deletion, and added docs
about how to safely uninstall HNC starting with the CRDs.

### CODE REVIEW

**Comment by adrianludwin** [comment]
/assign @yiqigao217 
cc @GinnyJI @adrianchung

**Comment by adrianludwin** [comment]
This addressed #718 as much as we're going to in HNC v0.5, so pushing this to the backlog.

File: incubator/hnc/docs/user-guide/how-to.md [hunk 1]
```diff
@@ -176,78 +176,80 @@ Kubernetes 1.15 and higher installed. See [issue
...
 These errors are there for your protection. Deleting namespaces is very
 dangerous, and deleting _subnamespaces_ can result in entire subtrees of
-namespaces being deleted as well. Therefore, you may set the
-`allowCascadingDelete` field either on the child namespace, on its parent, or (if
-the parent is a subnamespace itself) on its parent, and so on. For example, set
-the field on `child` if you’re only trying to delete the child, or on `parent` if
-you’re trying to delete the parent.
+namespaces being deleted as well. Therefore, if deleting a namespace (or
+subnamespace) would result in the deletion of any namespace _other than the one
+explicitly being deleted_, HNC requires that you must specify the
```
Comment (yiqigao217) [inline incubator/hnc/docs/user-guide/how-to.md:211]: This is documented well but I am just thinking about the easy deletion of leaf subns. Is it possible that users may delete a leaf subns' parent (full ns) without knowing the subns will be deleted too?

Comment (adrianludwin) [reply incubator/hnc/docs/user-guide/how-to.md:211]: That's not allowed:

```
aludwin@aludwin0:hnc$ k create ns parent
namespace/parent created

aludwin@aludwin0:hnc$ k hns create child -n parent
Successfully created "child" subnamespace anchor in "parent" namespace

aludwin@aludwin0:hnc$ k delete ns parent
Error from server (Forbidden): admission webhook "namespaces.hnc.x-k8s.io" denied the request: Please set allowCascadingDelete first either in the parent namespace or in all the subnamespaces.
  Subnamespace(s) without allowCascadingDelete set: [child].
```

Comment (yiqigao217) [reply incubator/hnc/docs/user-guide/how-to.md:211]: Ah right, thank you for clarification. Yes, I tested it and got the same. Only deleting the leaf subns anchor doesn't need allowCascadingDelete set.

**Pull request merged**